fs/dafsserver/dafsserver.cpp

```diff
@@ -5017,9 +5018,22 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
         unsigned posOfErr = reply.length();
         try
         {
-            switch(cmd)
+            /* isRowServiceClient only set for bare-metal clients
+             * Check they aren't sending a non-rowservice command to the row service
+             */
+            switch (cmd)
+            {
+                case RFCStreamGeneral:
+                case RFCStreamRead:
+                case RFCStreamReadJSON:
+                    break;
+                default:
+                    if (!directIO || client->isRowServiceClient())
+                        throw createDafsException(DAFSERR_cmdstream_unauthorized, "Unauthorized command");
+            }
```

Comment:
trivial: slightly better to have a break at the end of the default case.  (Some checkers may complain.)

Reply:
right, will add.